src/io/flutter/view/InspectorPanel.java

```diff
@@ -163,6 +168,14 @@ private InspectorService getInspectorService() {
   }
 
   void setActivate(boolean enabled) {
+    // TODO(jacobr): where is the right place to be caching the pub roots.
```

Comment:
I'd like to cache the pubRoots somewhere as I do use the pub roots every  time a node is styled to determine whether it should be bold or not. If there is an event I can listen for when pub roots change that is also reasonable. I assume pub roots can't change on a hot reload.

Reply:
Perhaps in the `FlutterApp`? Or at least something that's per-debug session. I think it's reasonable to assume that they won't change during a debug session.

Reply:
What happens if we have two sessions running at the same time? Does the inspector become unhappy?

You can test this by launching an app against one device and then a 2nd.

Reply:
Cached on the FlutterApp for now. Good point about having two session running at the same time. We definitely don't handle that case currently.